cmd/ci-reporter/report.go

```diff
@@ -0,0 +1,364 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/google/go-github/v34/github"
+	"golang.org/x/oauth2"
+)
+
+type requiredJob struct {
+	OutputName string
+	URLName    string
+}
+
+func main() {
+	boolPtr := flag.Bool("short", false, "a short report for mails and slack")
+	flag.Parse()
+
+	githubAPIToken := os.Getenv("GITHUB_AUTH_TOKEN")
+	if githubAPIToken == "" {
+		fmt.Printf("Please provide GITHUB_AUTH_TOKEN env variable to be able to pull cards from the github board")
+		os.Exit(1)
+	}
+
+	releaseVersion := os.Getenv("RELEASE_VERSION")
+
+	err := printCardsOverview(githubAPIToken, *boolPtr)
+	if err != nil {
+		fmt.Printf("error when querying cards overview, exiting: %v\n", err)
+		os.Exit(1)
+	}
+
+	printJobsStatistics(releaseVersion)
+}
+
+const (
+	newCards                = 4212817
+	underInvestigationCards = 4212819
+	observingCards          = 4212821
+	ciSignalBoardProjectID  = 2093513
+)
+
+type issueOverview struct {
+	url   string
+	id    int64
+	title string
+	sig   string
+}
+
+func printCardsOverview(token string, setShort bool) error {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	newCardsOverview, err := getCardsFromColumn(newCards, client, token)
+	if err != nil {
+		return err
+	}
+
+	investigationCardsOverview, err := getCardsFromColumn(underInvestigationCards, client, token)
+	if err != nil {
+		return err
+	}
+
+	observingCardsOverview, err := getCardsFromColumn(observingCards, client, token)
+	if err != nil {
+		return err
+	}
+
+	resolvedCards, err := findResolvedCardsColumns(client)
+	if err != nil {
+		return err
+	}
+
+	resolvedCardsOverview, err := getCardsFromColumn(resolvedCards, client, token)
+	if err != nil {
+		return err
+	}
+
+	printCards(setShort, groupByCards(newCardsOverview), groupByCards(investigationCardsOverview), groupByCards(observingCardsOverview), groupByCards(resolvedCardsOverview))
+
+	return nil
+}
+
+func findResolvedCardsColumns(client *github.Client) (int64, error) {
+	opt := &github.ListOptions{}
+	columns, _, err := client.Projects.ListProjectColumns(context.Background(), ciSignalBoardProjectID, opt)
+	if err != nil {
+		return 0, err
+	}
+
+	resolvedColumns := make([]*github.ProjectColumn, 0)
+	for _, v := range columns {
+		if v.Name != nil && *v.Name == "Resolved" {
+			resolvedColumns = append(resolvedColumns, v)
+		}
+	}
+
+	sort.Slice(resolvedColumns, func(i, j int) bool {
+		return resolvedColumns[i].GetID() < resolvedColumns[j].GetID()
+	})
+	return resolvedColumns[0].GetID(), err
+}
+
+func printCards(shortReport bool, newCards, investigation, observing, resolved map[string][]*issueOverview) {
```

Comment:
tried, it doesn't seem to work well in this case because when you copy and paste the report into Slack, the formatting is lost (see screenshot).

![slack-ci-signal-tablewriter-20211111191248-659x356](https://user-images.githubusercontent.com/58054077/141364523-5aa4ddec-24a9-4221-943f-2886f0f5069c.png)
s